tests/src/Functional/WorkbenchAccessTestTrait.php

```diff
@@ -41,41 +41,61 @@ public function setUpVocabulary() {
   }
 
   /**
-   * Sets up an access taxonomy field for a given vocab on a given node type.
+   * Sets up a taxonomy field on a given entity type and bundle.
    *
-   * @param \Drupal\node\Entity\NodeType $node_type
-   *   The node type to create the field on.
-   * @param \Drupal\taxonomy\Entity\Vocabulary $vocab
-   *   The vocab to create the field against.
+   * @param string $entity_type_id
+   *   Entity type ID.
+   * @param string $bundle
+   *   Bundle ID.
+   * @param string $vocabulary_id
+   *   Vocabulary ID.
+   * @param string $field_name
+   *   Field name.
    */
-  public function setUpTaxonomyField(NodeType $node_type, Vocabulary $vocab) {
-    // Create workbench access storage for node.
-    $field_storage = FieldStorageConfig::create([
-      'field_name' => WorkbenchAccessManagerInterface::FIELD_NAME,
-      'entity_type' => 'node',
-      'type' => 'entity_reference',
-      'cardinality' => 1,
-      'settings' => [
-        'target_type' => 'taxonomy_term',
-      ],
-    ]);
-    $field_storage->save();
-    // Create an instance of the access field on the content type.
-    FieldConfig::create([
-      'field_storage' => $field_storage,
-      'bundle' => $node_type->id(),
-      'settings' => [
-        'handler' => 'workbench_access:taxonomy_term',
-        'handler_settings' => [
-          'target_bundles' => [
-            $vocab->id() => $vocab->id(),
+  protected function setUpTaxonomyFieldForEntityType($entity_type_id, $bundle, $vocabulary_id, $field_name = WorkbenchAccessManagerInterface::FIELD_NAME) {
+    if (!$field_storage = FieldStorageConfig::load("$entity_type_id.$field_name")) {
```

Comment:
You can use `\Drupal\field\Tests\EntityReference\EntityReferenceTestTrait::createEntityReferenceField` here.